physical/raft/raft.go

```diff
@@ -274,7 +274,7 @@ func EnsurePath(path string, dir bool) error {
 	if !dir {
 		path = filepath.Dir(path)
 	}
-	return os.MkdirAll(path, 0o755)
+	return os.MkdirAll(path, 0o750)
```

Comment:
I remember seeing this as 0o700 in your main PR. https://github.com/hashicorp/vault/pull/14791/files